crates/notion-core/src/tool.rs

```diff
@@ -50,6 +50,25 @@ impl NotionFail for BinaryExecError {
     }
 }
 
+#[derive(Fail, Debug)]
+#[fail(display = "this tool is not yet implemented")]
+pub(crate) struct ToolUnimplementedError;
+
+impl ToolUnimplementedError {
+    pub(crate) fn new() -> Self {
+        ToolUnimplementedError {}
```

Comment:
Don't use the optional `{}` here -- you can just write `ToolUnimplementedError`.